File: perl/problem4.pl
use euler;

my $max = 0;
for my $x (100..999) {
	for my $y ($x..999) {
		$p = $x * $y;
		if ($p > $max && euler::ispalindrome($p))
		{
			$max = $p;
		}
	}
}

print $max . "\n";

File: python/problem4.py
import sys

def isPalindrome(n):
	if (len(n) % 2 != 0):
		return False;
	
	i = 0;
	while (i < len(n)/2):
		if (n[i] != n[ (len(n) - 1) - i]):
			return False;
		i = i + 1;

	return True;

i = 100;
maxp=0;
maxi=0;
maxj=0;

while (i <= 999):
	j = 100;
	while (j <= 999):
		if (isPalindrome(str(j * i))):
			print (i, j, j*i, sep=" ", end="\n");
			if (j * i > maxp):
				maxp = j * i;
				maxi = i;
				maxj = j;
		j = j + 1;
	i = i + 1;
	
print (maxi,"*",maxj,"=", maxp);

	




	